### PR TITLE
Add log SQUID code testing to redirect.test.py Au test.

### PR DESCRIPTION
**Pull request by ywkaras**

### CODE REVIEW

**Comment by ywkaras** [comment]
Marked this for 9.2 because it's very low risk, thus allowing easy backport of possible dependent commits.

File: tests/gold_tests/redirect/wait_for_log.sh [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+WAIT=120 # seconds
+LINES=17
+LOG_FILE=$1
+PORT=$2
+
+while ((WAIT > 0))
+do
+    if [[ -f $LOG_FILE ]] ; then
+        if (( $(wc -l < $LOG_FILE) >= LINES )) ; then
+            sed "s/:$PORT/:PORT/" $LOG_FILE
```
Comment (alficles) [inline tests/gold_tests/redirect/wait_for_log.sh:31]: These `$LOG_FILE`s need quotes or it'll break if the filename has a space in it for some crazy reason.

Comment (ywkaras) [reply tests/gold_tests/redirect/wait_for_log.sh:31]: Isn't that a wages of sin are death type thing?

Comment (ywkaras) [reply tests/gold_tests/redirect/wait_for_log.sh:31]: I suppose it's possible the directory path to the log file could have spaces.

Comment (ywkaras) [reply tests/gold_tests/redirect/wait_for_log.sh]: @alficles This is ready for re-review.

Comment (ywkaras) [reply tests/gold_tests/redirect/wait_for_log.sh]: > @alficles This is ready for re-review.

**Review by traeak — APPROVED** [review]
These changes all look fine.  I'm not so sure about the bash dependency but we have that all over the place in autests now as it is.

**Comment by ywkaras** [comment]
[approve ic autest]

**Comment by ywkaras** [comment]
[approve ci clang-format]

**Comment by ywkaras** [comment]
[approve ci autest]

**Comment by ywkaras** [comment]
[approve ci Clang-Analyzer]

**Comment by ywkaras** [comment]
> These changes all look fine. I'm not so sure about the bash dependency but we have that all over the place in autests now as it is.

Hmm what's the alternative you propose?  Only use Python for scripts?  That may be a good idea if it's guaranteed that Au test test runs will use the same Python version as Au test was run with.  Bash seems to have less issues with incompatible versions than Python.

**Pull request merged**